### PR TITLE
feat(argo-cd): Move argocd-cm and argocd-rbac-cm to config section

### PR DESCRIPTION
**Pull request by pdrastil**
Resolves: https://github.com/argoproj/argo-helm/issues/1519

Decouple argocd-cm and argocd-rbac-cm from server component and keep this together with rest of the configs. Backward compatibility ensured with coalesce and deprecation notice.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/argocd-configs/argocd-rbac-cm.yaml [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
     {{- include "argo-cd.labels" (dict "context" . "component" .Values.server.name "name" "rbac-cm") | nindent 4 }}
-  {{- with .Values.server.rbacConfigAnnotations }}
+  {{- with (coalesce .Values.server.rbacConfigAnnotations .Values.configs.rbac.annotations) }}
   annotations:
     {{- range $key, $value := . }}
     {{ $key }}: {{ $value | quote }}
     {{- end }}
   {{- end }}
-{{- with .Values.server.rbacConfig }}
+{{- with (coalesce .Values.server.rbacConfig (omit .Values.configs.rbac "create" "annotations")) }}
```
Comment (pdrastil) [inline charts/argo-cd/templates/argocd-configs/argocd-rbac-cm.yaml:14]: Coalesce makes this backward compatible change - Users can upgrade and then migrate to new section.

**Review by jmeridth — APPROVED** [review]
LGTM 👍

The `coalesce` allows for backwards compatibility.  Nice.

File: charts/argo-cd/templates/argocd-configs/argocd-cm.yaml [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-  {{- with .Values.server.configAnnotations }}
+  {{- with (coalesce .Values.server.configAnnotations .Values.configs.cm.annotations) }}
   annotations:
     {{- range $key, $value := . }}
     {{ $key }}: {{ $value | quote }}
     {{- end }}
   {{- end }}
 data:
-  {{- include "argo-cd.config" $ | nindent 2 }}
+  {{- include "argo-cd.config.cm" . | nindent 2 }}
```
Comment (stemar94) [inline charts/argo-cd/templates/argocd-configs/argocd-cm.yaml:15]: config -> config**s**

**Review by stemar94 — COMMENTED** [review]
Typo in templates/argocd-configs/argocd-cm.yaml

**Pull request merged**